### PR TITLE
Switch to LXD from LXC in juju2

### PR DESCRIPTION
**Pull request by makyo**

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5729/
Test PASSed.

File: jujugui/static/gui/src/test/test_bundle_importer.js [hunk 1]
```diff
@@ -379,8 +379,8 @@ describe('Bundle Importer', function() {
         assert.equal(db.machines.item(0).id, 'new0');
         assert.equal(db.machines.item(1).id, 'new1');
         assert.equal(db.machines.item(2).id, 'new4');
-        assert.equal(db.machines.item(3).id, 'new0/lxc/new3');
-        assert.equal(db.machines.item(4).id, 'new1/lxc/new2');
+        assert.equal(db.machines.item(3).id, 'new0/lxd/new3');
+        assert.equal(db.machines.item(4).id, 'new1/lxd/new2');
```
Comment (hatched) [inline jujugui/static/gui/src/test/test_bundle_importer.js]: We should have a test to make sure that Juju 1 still gets lxc

File: jujugui/static/gui/src/app/utils/bundle-importer.js [hunk 2]
```diff
@@ -391,10 +391,14 @@ YUI.add('bundle-importer', function(Y) {
...
+      let containerType = record.args[0].containerType;
+      if (containerType === 'lxc' && !this.isLegacyJuju) {
+        containerType = 'lxd';
+      }
       // XXX This code is duplicated from scale-up.js:191. We need to create a
       // layer where we create ghosts and handle cleaning them up.
       var machine = this.db.machines.addGhost(
-          record.args[0].parentId, record.args[0].containerType);
+          record.args[0].parentId, containerType);
       this.env.addMachines(record.args, function(machine) {
```
Comment (frankban) [inline jujugui/static/gui/src/app/utils/bundle-importer.js]: ISTM that we are passing the old record.args[0].containerType here, which is confusing.
Maybe we should just modify record.args in place? That way there would be no ambiguity and it's more future proof?
```javascript
if (record.args[0].containerType === 'lxc' && !this.isLegacyJuju) {
record.args[0].containerType = 'lxd';
```
What do you think?

File: jujugui/static/gui/src/test/test_bundle_importer.js (hunk 1 above)
Comment (frankban) [inline jujugui/static/gui/src/test/test_bundle_importer.js]: +1

**Comment by frankban** [comment]
Thanks for this fix Madison!
👍 with a couple of suggestions.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5731/
Test PASSed.

**Comment by hatched** [comment]
Thanks for putting this together @makyo :+1: :ok:

**Comment by makyo** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**